### PR TITLE
Iterator/Page find for !lib and !list

### PR DESCRIPTION
**Pull request by willi-on-fire**
New Iterator/Page tracker for !lib and !list command

Changed lib_to_string() and list_to_string() to reflect the page listing changes

Added a page size for bot to list the # of songs per page

Took out bot says from playlist and library
list size constant exists inside playlist and library

Changed shownext and showprev to nextsong and prevsong

!shutup and !speakfool is added. Not implemented because bot not chatting on each new song without manual activation

There is a play bug for asian chars in file name with the playlist.

changed skip to be similar to prev to dodge bug
manual advancing works now. old skip also had play bug.

### CODE REVIEW

File: library.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
+from os.path import splitext, basename
+import glob
+
+class Library:
+  def __init__(self, directory):
+    self.library = []
+    self.music_dir = directory
+    self.list_size = 20
```
Comment (azin634) [inline library.py]: this should be a constant instead

File: library.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    self.populate_lib()
+
+  def populate_lib(self):
```
Comment (azin634) [inline library.py]: can you rename this to populate()?

File: library.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    for f in glob.glob(self.music_dir+"*.mp3"):
+      song_name = splitext(basename(f))[0]
+      self.library.append(song_name)
+
+  def lib_contains_song(self, song:str):
```
Comment (azin634) [inline library.py]: can you rename this to contains_song?

File: library.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    return song in self.library
+
+  def get_library(self):
+    return list(self.library)
+
+  def get_page_count(self):
+    return int(len(self.library)/self.list_size + 1)
+
+  def lib_to_string(self, page: int):
```
Comment (azin634) [inline library.py]: can you rename this to to_string?

File: music_player.py [hunk 5]
```diff
@@ -139,19 +131,23 @@ def stop_player(self):
       await self.bot.say('THERE IS **NO** MUSIC PLAYING BITCH')
 
   @commands.command(no_pm=True)
-  async def lib(self):
-    await self.bot.say(self.lib_to_string())
+  async def lib(self, page: int):
+    string = self.library.lib_to_string(page)
```
Comment (azin634) [inline music_player.py]: I think this should be moved down to line 140, after the else

File: music_player.py [hunk 6]
```diff
@@ -168,9 +164,15 @@ def stop_player(self):
       await self.bot.say('__'+ song + '__ **DOESN\'T** EXIST IN PLAYLIST BITCH')
 
   @commands.command(no_pm=True)
-  async def list(self):
+  async def list(self, page: int):
     if not self.playlist.is_empty():
-      await self.bot.say(self.playlist.list_to_string())
+      string = self.playlist.list_to_string(page)
```
Comment (azin634) [inline music_player.py]: This should also probably happen after the else on line 174

File: playlist.py [hunk 7]
```diff
@@ -51,14 +63,24 @@ def rewind_playlist(self):
...
-
   def is_empty(self):
     return len(self.playlist) == 0
 
   def clear(self):
     del self.playlist[:]
 
-
-
+  def list_to_string(self, page: int):
```
Comment (azin634) [inline playlist.py]: can you rename this to_string?

**Comment by azin634** [comment]
Please push to this PR instead of making a new one so we can track the comments

File: library.py (hunk 1 above)
Comment (willi-on-fire) [inline library.py]: Addressing this in the next pull along with some other ones

**Pull request merged**